x-pack/plugins/uptime/public/components/overview/kuery_bar/typeahead/search_type/search_type.test.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { fireEvent } from '@testing-library/react';
+import { act } from 'react-dom/test-utils';
+import { render } from '../../../../../lib/helper/rtl_helpers';
+import { SearchType } from './search_type';
+
+describe('Kuery bar search type', () => {
+  it('can change from simple to kq;', () => {
+    let kqlSyntax = false;
+    const setKqlSyntax = jest.fn((val: boolean) => {
+      kqlSyntax = val;
+    });
+
+    const { getByTestId } = render(
+      <SearchType kqlSyntax={kqlSyntax} setKqlSyntax={setKqlSyntax} />
+    );
+
+    // open popover to change
+    fireEvent.click(getByTestId('syntaxChangeToKql'));
+
+    act(() => {
+      // change syntax
+      fireEvent.click(getByTestId('toggleKqlSyntax'));
```

Comment:
fireEvent is already wrapped in act https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily. If you're getting warnings, it's likely something else. If there's an async action, may need to wrap expects in `waitFor`